internal/controller/account/servicebinding/servicebinding.go

```diff
@@ -124,49 +161,157 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 	}
 
 	cr.SetConditions(xpv1.Creating())
-	if err := e.tfClient.Create(ctx); err != nil {
+
+	// Generate btpName if not already set in spec
+	var btpName string
+	if e.isRotationEnabled(cr) {
+		btpName = servicebindingclient.GenerateRandomName(cr.Spec.ForProvider.Name)
+	} else {
+		btpName = cr.Spec.ForProvider.Name
+	}
+	cr.Spec.BtpName = &btpName
+
+	// Update the spec with the generated btpName
+	if err := e.kube.Update(ctx, cr); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errUpdateStatus)
+	}
+
+	_, _, creation, err := e.instanceManager.CreateInstance(ctx, cr, *cr.Spec.BtpName)
+	if err != nil {
 		return managed.ExternalCreation{}, errors.Wrap(err, errCreateBinding)
 	}
 
-	return managed.ExternalCreation{
-		ConnectionDetails: managed.ConnectionDetails{},
-	}, nil
+	if err := e.kube.Status().Update(ctx, cr); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errUpdateStatus)
+	}
+
+	// Remove force rotation annotation after successful creation
+	if cr.ObjectMeta.Annotations != nil {
+		if _, ok := cr.ObjectMeta.Annotations[servicebindingclient.ForceRotationKey]; ok {
+			meta.RemoveAnnotations(cr, servicebindingclient.ForceRotationKey)
+		}
+	}
+
+	return creation, nil
 }
 
-func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
-	_, ok := mg.(*v1alpha1.ServiceBinding)
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.ServiceBinding)
 	if !ok {
 		return managed.ExternalUpdate{}, errors.New(errNotServiceBinding)
 	}
-	if err := c.tfClient.Update(ctx); err != nil {
-		return managed.ExternalUpdate{}, err
+
+	// Check if current binding is already retired - if so, skip update as service bindings are immutable
+	currentBindingRetired := false
+	for _, retiredKey := range cr.Status.AtProvider.RetiredKeys {
+		if retiredKey.ID == cr.Status.AtProvider.ID {
+			currentBindingRetired = true
+			break
+		}
 	}
-	return managed.ExternalUpdate{
-		ConnectionDetails: managed.ConnectionDetails{},
-	}, nil
+
+	// Only update if the current binding is not retired (service bindings are immutable in BTP)
+	var updateResult managed.ExternalUpdate
+	if !currentBindingRetired {
+		// Use btpName from spec, fall back to name for backward compatibility
```

Comment:
duplciated code and prone to be forgotten once we remove the old logic one day --> make a little heper func for that